### PR TITLE
podvm-mkosi: remove cloud-init

### LINKED ISSUE #1467: RFC: Remove cloud-config / cloud-init from PodVM Image?
# Problem Statement

Currently, cloud-config is running at VM startup to configure the system.

## How we use it

- If `DISABLE_CLOUD_CONFIG` is unset, we populate the fs with configuration files using the `write_files` directive.
- We use it to start CSP-specific agents, like Azure's `waagent`.
- Potentially network configuration

## Why we might not want to use it

Cloud Config is a venue to [execute](https://cloudinit.readthedocs.io/en/latest/reference/examples.html#run-commands-on-first-boot) arbitrary untrusted code at startup.

# Suggested Solution

Replace the required functionality (config file creation, cloud callback, network configuration) with systemd units. We started this work by providing an option to write config files by parsing user-data directly (`DISABLE_CLOUD_CONFIG=true`).

# Alternative Solutions

- Measure cloud-config data at early boot stages, prior to processing.

# Open Problems

I'm not sure whether alternatives to cloud-config are available. I think in some cases a virtual iso is attached and we'd need to support that one as a source in addition to `process-user-data`.

### PR DESCRIPTION
**Pull request by katexochen**
This is based on #1524, which should be reviewed and merged first.

Fixes #1467 (for x86, we should check with the IBM folks if they want to remove cloud init for s390x as well)

This requires `DISABLE_CLOUD_CONFIG=true` in your `install/overlay/${CSP}/kustomization.yaml`

### CODE REVIEW

File: pkg/adaptor/cloud/azure/provider.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func (p *azureProvider) CreateInstance(ctx context.Context, podName, sandboxID s
 
 	result, err := p.create(ctx, vmParameters)
 	if err != nil {
-		if err := p.deleteDisk(ctx, diskName); err != nil {
+		if err := p.deleteDisk(context.Background(), diskName); err != nil {
```
Comment (mkulke) [inline pkg/adaptor/cloud/azure/provider.go:235]: unrelated?

File: podvm-mkosi/README.md [hunk 2]
```diff
@@ -65,5 +65,5 @@ Packages=
 The following limitations apply to these images. Notice that the limitations are intentional to
 reduce complexity of configuration and CI and shall not be seen as open to-dos.
 
-- `DISABLE_CLOUD_CONFIG=false` is implied. The mkosi images are currently using
-    cloud init/cloud config, and there is no possibility to disable it.
+- `DISABLE_CLOUD_CONFIG=true` is implied. The mkosi images are using
+    our own mechanism to fetch metadata from the IMDS API. Cloud init is not supported.
```
Comment (mkulke) [inline podvm-mkosi/README.md:69]: I'd add the note from the PR description on how to disable it.

File: pkg/adaptor/cloud/azure/provider.go (hunk 1 above)
Comment (katexochen) [reply pkg/adaptor/cloud/azure/provider.go:235]: Yes, thought this is to minor for a separate PR, but can cherry-pick if you like.

File: podvm-mkosi/mkosi.presets/system/mkosi.conf.d/fedora-debug.conf [hunk 3]
```diff
@@ -15,3 +15,6 @@ Packages=
     strace
     openssh-server
     file
+    iputils
+    curl
+    wget
```
Comment (mkulke) [inline podvm-mkosi/mkosi.presets/system/mkosi.conf.d/fedora-debug.conf:20]: unrelated?

**Review by mkulke — APPROVED** [review]
thanks, tested successfully

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @katexochen
